### ISSUE TITLE
Failed to move CFD to closed CFDs

### ISSUE THREAD

**Issue by da-kami** [itchysats/itchysats#2086]
Source: taker mainnet database on Umbrel.

There appears to be another problem with building the CET:

```
2022-05-23 00:31:51 DEBUG daemon::db: Applying new events to CFD order_id=6236074d-44e7-41ec-aec6-121b30b682ac aggregate=daemon::projection::Cfd cfd_version=0 num_events=1743
2022-05-23 00:31:51 DEBUG daemon::db: Applying new events to CFD order_id=6236074d-44e7-41ec-aec6-121b30b682ac aggregate=daemon::oracle::Cfd cfd_version=0 num_events=1743
2022-05-23 00:31:51 TRACE daemon::oracle: Skipping /x/BitMEX/BXBT/2022-05-24T00:00:00.price?n=20 because it likely hasn't occurred yet
2022-05-23 00:31:51 DEBUG daemon::db: Applying new events to CFD order_id=6236074d-44e7-41ec-aec6-121b30b682ac aggregate=daemon::monitor::Cfd cfd_version=0 num_events=1743
2022-05-23 00:31:51 DEBUG daemon::oracle: Fetching attestation for /x/BitMEX/BXBT/2022-05-11T01:00:00.price?n=20
2022-05-23 00:31:51 DEBUG daemon::db: Applying new events to CFD order_id=6236074d-44e7-41ec-aec6-121b30b682ac aggregate=daemon::position_metrics::Cfd cfd_version=0 num_events=1743
2022-05-23 00:31:51 DEBUG daemon::db: Applying new events to CFD order_id=6236074d-44e7-41ec-aec6-121b30b682ac aggregate=model::cfd::Cfd cfd_version=0 num_events=1743
2022-05-23 00:31:51 TRACE daemon::auto_rollover: CFD is not eligible for auto-rollover order_id=6236074d-44e7-41ec-aec6-121b30b682ac reason=Cannot roll over when CFD is already closed
2022-05-23 00:31:51  WARN daemon::db::closed: Failed to move closed CFD: Missing spend script in CET: script pubkey not found in tx order_id=6236074d-44e7-41ec-aec6-121b30b682ac
```

**Comment by da-kami** [comment]
Also seen on `testnet` taker with an older CFD:

```
2022-05-23 02:43:15  WARN daemon::db::closed: Failed to move closed CFD: Missing spend script in CET: script pubkey not found in tx order_id=6236074d-44e7-41ec-aec6-121b30b682ac
```

**Comment by da-kami** [comment]
Note on the ticket: 

`6236074d-44e7-41ec-aec6-121b30b682ac` was a position where collab close failed. When force closing we ran into a bug where the CET was wrongly constructed because of a `0` `TxOut`. This was fixed.
On the taker side the collab close event was removed from the database and the CET was published (so funds are not locked forever) upon startup. CET on chain: https://mempool.space/tx/c1c376c4f2ca95d324a765302191ec84469df46770679c837bb5da38eb0c2d86

However, when the taker tries to move this CFD into closed it runs into an error, so there must be some bug somewhere. This bug is likely just when moving the CFD to closed. 

Note: The taker UI currently displays that this CFD was refunded (which is incorrect). This is likely just a problem in the projection aggregate and should not be relevant for this ticket, see https://github.com/itchysats/itchysats/issues/2168
